src/DestroyStrategy.groovy

```diff
@@ -0,0 +1,53 @@
+import static TerraformEnvironmentStage.ALL
+import static TerraformEnvironmentStage.PLAN
+import static TerraformEnvironmentStage.CONFIRM
+import static TerraformEnvironmentStage.DESTROY
+
+class DestroyStrategy {
+
+    private TerraformInitCommand initCommand
+    private TerraformPlanCommand planCommand
+    private TerraformApplyCommand destroyCommand
+    private Jenkinsfile jenkinsfile
+
+    public Closure createPipelineClosure(String environment, StageDecorations decorations) {
+        initCommand = TerraformInitCommand.instanceFor(environment)
+        planCommand = TerraformPlanCommand.instanceFor(environment)
+        planCommand = planCommand.withArgument("-destroy")
+        destroyCommand = TerraformApplyCommand.instanceFor(environment)
+        destroyCommand = destroyCommand.withCommand("destroy")
+        jenkinsfile = Jenkinsfile.instance
+
+        return { ->
+            node(jenkinsfile.getNodeName()) {
+                deleteDir()
+                checkout(scm)
+
+                decorations.apply(ALL) {
+                    stage("${PLAN}-${DESTROY}-${environment}") {
+                        decorations.apply(PLAN) {
+                            sh initCommand.toString()
+                            sh planCommand.toString()
+                        }
+                    }
+
+                    decorations.apply("Around-${CONFIRM}") {
+                        stage("${CONFIRM}-${DESTROY}-${environment}") {
+                            decorations.apply(CONFIRM) {
+                                echo "Approved"
+                            }
+                        }
+                    }
+
+                    decorations.apply("Around-${DESTROY}") {
+                        stage("${DESTROY}-${environment}") {
+                            decorations.apply(DESTROY) {
+                                sh destroyCommand.toString()
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
```

Comment:
Same "If we only attack UseCase 1 in this PR (destroy-only), we no longer need these changes."